Install yamllint via pipx to get more modern version, also run ensurepath as the runner user

pdewilde opened the pull request:
This solves two issues:

1. We were installing yamllint via apt. This was pulling in an old verison, and is different to how it is done in https://github.com/actions/runner-images?tab=readme-ov-file#package-managers-usage

2. `/home/runner/.local/bin` wasn't being added to the path, so installing newer versions of yamllint didn't cause the newer version to be used.

drevell reviewed (changes requested):
There are some interesting check warnings about things that look important

pdewilde commented:
> There are some interesting check warnings about things that look important

Fixed some of them. Some intentionally left unfixed:

"Last USER should not be root" --> I intend the setup-docker to be run as root, as it cannot be done at image build time. I then degrade to runner user using gosu. I could run all those commands with sudo instead tough.

"Pin version in apt get install" --> I don't want to do this now as we are going to me moving to a newer base image soon. Generally I'm not sure if we want to, otherwise we wouldn't consume any security updates and patches provided by the distro maintainer unless we manually went through and looked if newer versions existed.

"Alsways tag the version of an image explicitly" --> I've looked through github.com/abcxyz and don't see any other instances of us doing that. I did it for the image file I changed, but am not certain if that is a pattern we follow.
